components/fxa-client/src/internal/util.rs

```diff
@@ -52,3 +52,27 @@ impl Xorable for [u8] {
         }
     }
 }
+
+pub fn parse_url(url: &str, when: impl Into<String>) -> Result<url::Url> {
+    url::Url::parse(url).map_err(|_| Error::MalformedUrl {
+        sanitized_url: sanitized_url(url),
+        when: when.into(),
+    })
+}
+
+pub fn join_url(url: &url::Url, path: &str, when: impl Into<String>) -> Result<url::Url> {
+    url.join(path).map_err(|_| Error::MalformedUrl {
+        sanitized_url: sanitized_url(url.as_str()),
+        when: when.into(),
+    })
+}
+
+fn sanitized_url(url: &str) -> String {
+    // Remove everything after the `?` char, this is the URL params where all the auth data
+    // goes.
+    match url.split_once("?") {
```

Comment:
This is non-blocking since i'm not entirely sure, but is it possible for fragments to occur before the query? if so we can maybe include `split_once(['?','#'])` but if not ignore.

Reply:
Great point!  I think they should go after the query, but there's also the possibility of only a fragment.  I'm going to update the code to match your suggestion.